spring-integration-redis/src/test/java/org/springframework/integration/redis/util/RedisLockRegistryTests.java

```diff
@@ -692,6 +692,63 @@ public void multiRedisLockRegistryTest() throws InterruptedException, ExecutionE
 	}
 
 
+	@Test
+	@RedisAvailable
+	public void earlyWakeUpTest() throws InterruptedException {
+		final int THREAD_CNT = 3;
+		final String testKey = "testKey";
+
+		final CountDownLatch tryLockReady = new CountDownLatch(2);
+		final CountDownLatch awaitTimeout = new CountDownLatch(2);
+		final RedisConnectionFactory connectionFactory = getConnectionFactoryForTest();
+		final RedisLockRegistry registry1 = new RedisLockRegistry(connectionFactory, this.registryKey);
+		final RedisLockRegistry registry2 = new RedisLockRegistry(connectionFactory, this.registryKey);
+		final RedisLockRegistry registry3 = new RedisLockRegistry(connectionFactory, this.registryKey);
+		final ExecutorService executorService = Executors.newFixedThreadPool(THREAD_CNT);
```

Comment:
I think the `ExecutorService` has to be `shutdown()` in the end of test method to free resources.